cmd/svcat/broker/register_cmd.go

```diff
@@ -18,57 +18,122 @@ package broker
 
 import (
 	"fmt"
+	"os"
+	"strings"
+	"time"
 
 	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/command"
 	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/output"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	servicecatalog "github.com/kubernetes-incubator/service-catalog/pkg/svcat/service-catalog"
 	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // RegisterCmd contains the information needed to register a broker
 type RegisterCmd struct {
-	BrokerName string
-	Context    *command.Context
-	URL        string
+	*command.Waitable
+
+	Context *command.Context
+
+	BasicSecret       string
+	BearerSecret      string
+	BrokerName        string
+	CAFile            string
+	ClassRestrictions []string
+	PlanRestrictions  []string
+	SkipTLS           bool
+	RelistBehavior    string
+	RelistDuration    time.Duration
+	URL               string
 }
 
 // NewRegisterCmd builds a "svcat register" command
 func NewRegisterCmd(cxt *command.Context) *cobra.Command {
 	registerCmd := &RegisterCmd{
-		Context: cxt,
+		Context:  cxt,
+		Waitable: command.NewWaitable(),
 	}
 	cmd := &cobra.Command{
-		Use:   "register NAME --url URL",
+		Use:   "register NAME URL",
 		Short: "Registers a new broker with service catalog",
 		Example: command.NormalizeExamples(`
-		svcat register mysqlbroker --url http://mysqlbroker.com
+		svcat register mysqlbroker http://mysqlbroker.com
 		`),
 		PreRunE: command.PreRunE(registerCmd),
 		RunE:    command.RunE(registerCmd),
 	}
-	cmd.Flags().StringVar(&registerCmd.URL, "url", "",
-		"The broker URL (Required)")
-	cmd.MarkFlagRequired("url")
+	cmd.Flags().StringVar(&registerCmd.BasicSecret, "basic-secret", "",
+		"A secret containing basic auth (username/password) information to connect to the broker")
+	cmd.Flags().StringVar(&registerCmd.BearerSecret, "bearer-secret", "",
+		"A secret containing a bearer token to connect to the broker")
+	cmd.Flags().StringVar(&registerCmd.CAFile, "ca", "",
+		"A file containing the CA certificate to connect to the broker")
+	cmd.Flags().StringSliceVar(&registerCmd.ClassRestrictions, "class-restrictions", []string{},
+		"A whitelist of Classes to allow from the broker")
+	cmd.Flags().StringSliceVar(&registerCmd.PlanRestrictions, "plan-restrictions", []string{},
+		"A whitelist of Plans to allow from the broker")
+	cmd.Flags().StringVar(&registerCmd.RelistBehavior, "relist-behavior", "",
+		"Behavior for relisting the broker's catalogue. Manual/Duration")
+	cmd.Flags().DurationVar(&registerCmd.RelistDuration, "relist-duration", 0*time.Second,
```

Comment:
If we send 0s as the default, and someone does `svcat register broker --relist-duration duration` without specifying a `--relist-duration` flag, does 0s end up being used? 

I suggest that we default to a recommended duration, not sure what that should be off the top of my head though. Mostly because it's really not clear what the behavior will end up being if we stick with 0s.